.github/workflows/ci.yaml

```diff
@@ -0,0 +1,165 @@
+name: Weekly global Tool Linting and Tests
+on:
+  schedule:
+    # Run at midnight every monday
+    - cron: '0 0 * * 1'
+  repository_dispatch:
+    types: [run-all-tool-tests-command]
+env:
+  GALAXY_RELEASE: release_20.09
+jobs:
+  setup:
+    name: Setup cache
+    if: github.repository_owner == 'TGAC'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7]
+    steps:
+    - name: Add reaction
+      if: ${{ github.event.client_payload.slash_command.command == 'run-all-tool-tests' }}
+      uses: peter-evans/create-or-update-comment@v1
+      with:
+        token: ${{ secrets.PAT }}
+        repository: ${{ github.event.client_payload.github.payload.repository.full_name }}
+        comment-id: ${{ github.event.client_payload.github.payload.comment.id }}
+        reaction-type: hooray
+    - uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Cache .cache/pip
+      uses: actions/cache@v2
+      id: cache-pip
+      with:
+        path: ~/.cache/pip
+        key: pip_cache_${{ matrix.python-version }}_${{ env.GALAXY_RELEASE }}
+    # Install the `wheel` package so that when installing other packages which
+    # are not available as wheels, pip will build a wheel for them, which can be cached.
+    - name: Install wheel
+      run: pip install wheel
+    - name: Install Planemo
+      run: pip install planemo
+    - name: Fake a planemo run to update cache
+      run: |
+        touch tool.xml
+        PIP_QUIET=1 planemo test --galaxy_python_version ${{ matrix.python-version }} --no_conda_auto_init --galaxy_branch $GALAXY_RELEASE
+  test:
+    name: Test tools
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+    needs: setup
+    strategy:
+      fail-fast: false
+      matrix:
+        chunk: [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39]
```

Comment:
40 chunks is very much overkill for this repository. Also 4 chunks for `pr.yaml` is probably too much. It would be really cool if we could generate this part of the matrix according to the number of tools to test using something like: https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#example-6

Reply:
True. For now I opened an issue at IUC.